src/core/Object3D.js

```diff
@@ -303,44 +305,17 @@ class Object3D extends EventDispatcher {
 
 	}
 
-	add( object ) {
```

Comment:
Sorry for the late review!

Would you mind undo all changes related to `add()` in this PR? It really should focus on `attach()` and not on other methods. If we change `add()` in this way, we also need to update `remove()` and this is too much for a single PR.

The suggested version of `attach()` is fine. Inlining `add()` is ideal for fixing the event dispatching issue.

The optimization for #26740 is maybe not really necessary since `attach()` is normally used with objects which are already part of the scene graph. That means they already have a parent and thus use the existing code path not the new optimized one. In order to keep the method simple as possible, I would leave this bit out for now.